protocol/xmrmaker/claim.go

```diff
@@ -235,8 +235,20 @@ func (s *swapState) claimWithRelay() (*ethtypes.Receipt, error) {
 	if err != nil {
 		log.Warnf("failed to relay with DHT-advertised relayers: %s", err)
 		log.Infof("falling back to swap counterparty as relayer")
-		return s.relayClaimWithXMRTaker()
+		receipt, err = s.relayClaimWithXMRTaker()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Save the relayer fee to the database
+	s.info.RelayerFee = coins.RelayerFeeWei
```

Comment:
The Info data structure is accessed by multiple threads. This go process executing the code here is the only thread that changes the data structure, so it doesn't require a read locks to always get correctly synchronized data, but it needs to grab the write lock (introduce recently in https://github.com/AthanorLabs/atomic-swap/pull/471) before changing data.  Create a setter function on the Info type that grabs the lock for writing before updating the value.  Other threads only read the data structure via `DeepCopy`, which grabs the read lock, so no additional changes are required.

Reply:
If you use `RelayerFeeETH` above instead of `RelayerFeeWei`, you might make some math with the expected amount (which is in ETH) easier.